### PR TITLE
Check Existence of Artifact based on DynamoDB Entry

### PR DESCRIPTION
**Pull request by meshde**
* Create a DynamoDB Table:
   * TableName: `SubmissionArtifactMap`
   * Keys:
       * `submissionId` (Hash Key)
       * `artifactFileName` (Range Key)
   * Non-Key Attributes:
       * `s3Path`
* Insert entry into `SubmissionArtifactMap` when creating artifacts.
* Check if entry exists in `SubmissionArtifactMap` when checking for existence of an artifact
* Use the path `:challengeId/:memberId/:submissionId/artifacts/:filename` for uploading artifacts to S3.

### CODE REVIEW

**Comment by callmekatootie** [comment]
Ticket #170

**Comment by callmekatootie** [comment]
@rootelement A question - should the artifacts api be backward compatible? Considering that the location of the artifact will now change, this will affect older artifacts. Should we handle those?

File: src/services/ArtifactService.js [hunk 1]
```diff
@@ -34,6 +37,56 @@ function * _uploadToS3 (file, name) {
...
+/*
+ * Function to create a submission-artifact mapping
+ * @param {Object} mapObject The mapping object to be inserted
+ * @param {Object} parentSpan The parent span object
+ * @return {Promise}
+ */
+function * createSubmissionArtifactMap (mapObject, parentSpan) {
+  const createSubmissionArtifactMapSpan = tracer.startChildSpans('ArtifactService.createSubmissionArtifactMap', parentSpan)
+  createSubmissionArtifactMapSpan.setTag('submissionId', mapObject.submissionId)
+  createSubmissionArtifactMapSpan.setTag('artifactFileName', mapObject.artifactFileName)
```
[the remark this answers is not in the document]
Comment (meshde) [reply src/services/ArtifactService.js]: Updated.

**Comment by meshde** [comment]
@callmekatootie the following points have been addressed. Please have a look.

> 1. You missed updated list artifacts. It should query the table now and not S3
> 2. When an artifact is deleted, its mapping is not deleted from the table.

**Comment by rootelement** [comment]
@callmekatootie i spoke to @cwdcwd and we decided it should be backwards compatible.

**Pull request merged**